### PR TITLE
picture plugin

### PR DESCRIPTION
**Pull request by slayernominee**
addressing https://github.com/OlegWock/anori/issues/52

Bildschirmfoto 2024-04-03 um 21 22 51

the preview image (deer) is locally added. 

the image can be set via an url setting 
Bildschirmfoto 2024-04-03 um 21 44 05

### CODE REVIEW

**Comment by slayernominee** [comment]
the deer is btw from unsplash

File: src/plugins/picture/picture-plugin.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            </div>
+
+        <Button className="save-config" onClick={onConfirm}>Save</Button>
+    </div>);
+};
+
+const PicturePlugin = ({ config, instanceId }: WidgetRenderProps<PicturePluginWidgetConfigType>) => {
+    const deerExampleImage = browser.runtime.getURL('/assets/images/picturePreview.jpg')
+    return (<div className="PictureWidget">
+        <img className="Image" src={config.url !== '' ? config.url : deerExampleImage} />
```
Comment (OlegWock) [inline src/plugins/picture/picture-plugin.tsx:39]: I don't think we need this deer image at all. I think you can just set src like this `src={config.url}`

File: src/plugins/picture/picture-plugin.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    configurationScreen: PictureConfigScreen,
+    mainScreen: PicturePlugin,
+    mock: () => {
+        const deerExampleImage = browser.runtime.getURL('/assets/images/picturePreview.jpg')
+        return (<PicturePlugin instanceId="mock" config={{
+            url: deerExampleImage
+        }} />)
+    },
+    appearance: {
```
Comment (OlegWock) [inline src/plugins/picture/picture-plugin.tsx:56]: Could you add `withoutPadding: true` to the `appearance`? This way there won't be paddings around picture and that will look better imo

**Review by OlegWock — CHANGES_REQUESTED** [review]
Hello! Thanks for the PR, I added couple of comments

**Comment by slayernominee** [comment]
i see one problem with not using a placeholder image: 
with no internet connection the mock will not have an image to show / shows the alt text
respectively with a slow internet connection it shows nothing at first 

possible solutions for this would be if we either use a placeholder image (or maybe even some random selected one from a multiple intern images) or some css animation ig

**Comment by OlegWock** [comment]
That's true, but I think it will be a rare occurrence. Anori is browser extension, so user probably will have internet connection while using it anyway.  And even if they don't, picture widgets in folder won't work without internet anyway, so I don't think we need 'special treatment' for mock (which user will see a lot less often)

**Comment by slayernominee** [comment]
removed it

**Comment by OlegWock** [comment]
Thanks :)

**Pull request merged**